llvm/lib/Target/DirectX/DXILTranslateMetadata.cpp

```diff
@@ -300,6 +301,36 @@ static MDTuple *emitTopLevelLibraryNode(Module &M, MDNode *RMD,
   return constructEntryMetadata(nullptr, nullptr, RMD, Properties, Ctx);
 }
 
+// TODO: We might need to refactor this to be more generic,
+// in case we need more metadata to be replaced.
+static void translateBranchMetadata(Module &M) {
+  for (auto &F : M) {
+    for (auto &BB : F) {
+      auto *BBTerminatorInst = BB.getTerminator();
+
+      auto *HlslControlFlowMD =
+          BBTerminatorInst->getMetadata("hlsl.controlflow.hint");
+
+      if (!HlslControlFlowMD || HlslControlFlowMD->getNumOperands() < 2)
```

Comment:
Can the operands be `< 2` and this still be valid? That seems like something that should be an assert.

Reply:
If operands are `< 2`, the branch has no attributes associated with it, an assert would be triggered every time we have a branch with no attribute.

Reply:
If you have a better way to check this default case, please let me know your thoughts, I am not happy with the way this check current works either

Reply:
Huh? `getMetadata` returns `null` on failure (hence the null check). If it finds the named metadata, it must have two operands right? You even have this asserting on the SPIR-V side.